src/main/java/org/xmldb/api/security/Permissions.java

```diff
@@ -0,0 +1,357 @@
+ * following acknowledgment: "This product includes software developed by the XML:DB Initiative
+ * (http://www.xmldb.org/)." Alternately, this acknowledgment may appear in the software itself, if
+ * and wherever such third-party acknowledgments normally appear.
+ *
+ * 4. The name "XML:DB Initiative" must not be used to endorse or promote products derived from this
+ * software without prior written permission. For written permission, please contact info@xmldb.org.
+ *
+ * 5. Products derived from this software may not be called "XML:DB", nor may "XML:DB" appear in
+ * their name, without prior written permission of the XML:DB Initiative.
+ *
+ * THIS SOFTWARE IS PROVIDED ``AS IS'' AND ANY EXPRESSED OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE APACHE SOFTWARE FOUNDATION OR ITS CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT
+ * NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR
+ * BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ * =================================================================================================
+ * This software consists of voluntary contributions made by many individuals on behalf of the
+ * XML:DB Initiative. For more information on the XML:DB Initiative, please see
+ * <https://github.com/xmldb-org/>
+ */
+package org.xmldb.api.security;
+
+import static java.lang.Boolean.TRUE;
+import static org.xmldb.api.security.Permission.GROUP_EXECUTE;
+import static org.xmldb.api.security.Permission.GROUP_READ;
+import static org.xmldb.api.security.Permission.GROUP_WRITE;
+import static org.xmldb.api.security.Permission.OTHERS_EXECUTE;
+import static org.xmldb.api.security.Permission.OTHERS_READ;
+import static org.xmldb.api.security.Permission.OTHERS_WRITE;
+import static org.xmldb.api.security.Permission.OWNER_EXECUTE;
+import static org.xmldb.api.security.Permission.OWNER_READ;
+import static org.xmldb.api.security.Permission.OWNER_WRITE;
+import static org.xmldb.api.security.Permission.SET_GID;
+import static org.xmldb.api.security.Permission.SET_UID;
+import static org.xmldb.api.security.Permission.STICKY_BIT;
+
+import java.util.EnumMap;
+import java.util.EnumSet;
+import java.util.Set;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Parser for {@link Permission} values.
+ * 
+ * This class provides static methods to parse string and integer based permission modes into a set
+ * of permissions.
+ */
+public final class Permissions {
+  private static final char READ = 04;
```

Comment:
I am a little unsure why these are `char`?

Reply:
well just to use less memory... could be even of type `byte`

Reply:
I doubt in practice there will be any difference in memory use here ;-)